### PR TITLE
ENG-15747: Fail license check if node count exceeds number specified …

### PR DESCRIPTION
**Pull request by manjujames**
…in license, for enterprise license as well.

### CODE REVIEW

File: src/frontend/org/voltdb/utils/MiscUtils.java [hunk 1]
```diff
@@ -383,18 +383,10 @@ public static boolean validateLicense(LicenseApi licenseApi,
...
-                        " nodes, starting cluster with " + numberOfNodes + " nodes.");
-                valid = false;
-            }
-            // Trial, Pro & AWS licenses have a hard enforced limit
-            else {
-                hostLog.fatal("Warning, VoltDB license for a " + licenseApi.maxHostcount() + " node " +
-                        "attempted for use with a " + numberOfNodes + " node cluster.");
-                return false;
-            }
+            hostLog.fatal("Attempting to start with too many nodes (" + numberOfNodes + "). " +
```
Comment (JadenW) [inline src/frontend/org/voltdb/utils/MiscUtils.java]: This seems enforce on the start and recovery cases, but not in the elastic add case.
So there might be a case that user will be able to grow the cluster to number of hosts exceeding licensed, but will not be able to recovery afterwards. Is this an expected behavior?
Also, those enforcements are in clear code?

Comment (manjujames) [reply src/frontend/org/voltdb/utils/MiscUtils.java]: That's a good point. I will check with Ruth.

Comment (manjujames) [reply src/frontend/org/voltdb/utils/MiscUtils.java]: Yes, the enforcements are in clear code as of now.

Comment (manjujames) [reply src/frontend/org/voltdb/utils/MiscUtils.java]: Ruth opened a separate ticket to handle elastic add case. So I will merge this one.

**Pull request merged**